pkg/collectors/pod_test.go

```diff
@@ -98,311 +114,381 @@ func TestPodCollector(t *testing.T) {
 		# HELP kube_pod_spec_volumes_persistentvolumeclaims_readonly Describes whether a persistentvolumeclaim is mounted read only.
 		# TYPE kube_pod_spec_volumes_persistentvolumeclaims_readonly gauge
 	`
-	cases := []struct {
+
+	type cs_to_prom func(cs v1.ContainerStatus, ns string, pod string) string
+
+	pod_to_prom := func(p v1.Pod, fn_cs cs_to_prom, fn_ics cs_to_prom) string {
```

Comment:
on the hindsight, refactoring this code in these methods may have made it more complex. But it makes changing things around a breeze and adding more pods/containers.